### ISSUE TITLE
Support multipart/form-data

### ISSUE THREAD

**Issue by kgriffs** [falconry/falcon#953]
Depends on: #418 

* Add support to `Request.form` for loading non-file control name/value pairs from a `multipart/form-data` request.
* Add a new property, `files` to `Request`. This should provide an interface for reading one or more files submitted via an HTML form post. We should look at the standard library, werkzeug, and bottle for inspiration re the implementation. There may also be some standalone packages that may be helpful (search PyPI).

**Comment by lwcolton** [comment]
@kgriffs To be clear are you asking for a new multipart form parser implementation, and not to re-use cgi.FieldStorage?  https://github.com/python/cpython/blob/2.7/Lib/cgi.py#L352

**Comment by kgriffs** [comment]
@lwcolton I'm not necessarily opposed to using cgi.FieldStorage but we should see if there are any advantages to forking code from werkzeug, for example.

**Comment by kgriffs** [comment]
See also: #1236

**Comment by kgriffs** [comment]
Note: This will be part of native HTML form post handling work that will have received a sponsorship to fund. @vytas7 will likely be overseeing this work stream.

**Comment by vytas7** [comment]
Resolved in https://github.com/falconry/falcon/pull/1590

I'll create a follow-up issue for ASGI+multipart.
